packages/components/forms/src/text-input/TextInput.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useCallback, useState, ChangeEvent } from 'react';
+import { BaseInput } from '@contentful/f36-inputs';
+import { Box, Flex } from '@contentful/f36-core';
+import { Label } from '@contentful/f36-forms';
+import { ValidationMessage } from '@contentful/f36-validation-message';
+import { HelpText } from '@contentful/f36-helptext';
+import { CopyButton } from '@contentful/f36-copybutton';
+import getStyles from './TextInput.styles';
+import { TextInputProps } from './types';
+
+export const _TextInput = (
+  {
+    className,
+    testId = 'cf-ui-text-field',
+    countCharacters = false,
+    isStandalone = false,
+    isRequired = false,
+    helpText,
+    label,
+    id,
+    link,
+    validationMessage,
+    value,
+    onChange,
+    placeholder,
+    maxLength,
+    isInvalid,
+    isDisabled,
+    isReadOnly,
+    withCopyButton,
+    onCopy,
+    name,
+    icon,
+    ...otherProps
+  }: TextInputProps,
+  ref: React.Ref<HTMLInputElement>,
+) => {
+  const [valueState, setValueState] = useState<string | undefined>(value);
+  const styles = getStyles({ isInvalid, isDisabled });
+
+  // Store a copy of the value in state.
+  // This is used by this component when the `countCharacters`
+  // option is on
+  const handleOnChange = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      setValueState(e.target.value);
+      if (onChange) onChange(e);
+    },
+    [onChange],
+  );
+
+  const inputWithCopyButton = (
+    <Flex>
+      <BaseInput
+        testId={testId}
+        {...otherProps}
+        ref={ref}
+        label={label}
+        type="text"
+        onChange={handleOnChange}
+        placeholder={placeholder}
+        as="input"
+        name={name}
+        isInvalid={isInvalid}
+        isDisabled={isDisabled}
+        isReadOnly={isReadOnly}
+        icon={icon}
+        maxLength={maxLength}
+        className={styles.inputWithCopyButton}
+      />
+      <CopyButton
+        value={valueState}
+        onCopy={onCopy}
+        className={styles.copyButton}
+      />
+    </Flex>
+  );
+  return (
+    <Flex flexDirection="column" fullWidth>
+      {!isStandalone && (
+        <Flex justifyContent="space-between" alignItems="center" fullWidth>
+          <Label htmlFor={id} required={isRequired}>
+            {label}
+          </Label>
+          {link && link}
+        </Flex>
+      )}
+      {withCopyButton ? (
+        inputWithCopyButton
+      ) : (
+        <BaseInput
+          testId={testId}
+          {...otherProps}
+          ref={ref}
+          label={label}
+          type="text"
+          onChange={handleOnChange}
+          placeholder={placeholder}
+          as="input"
+          name={name}
+          isInvalid={isInvalid}
+          isDisabled={isDisabled}
+          isReadOnly={isReadOnly}
+          icon={icon}
+          maxLength={maxLength}
+        />
+      )}
+
+      {!isStandalone && (
+        <>
+          {(helpText || countCharacters || validationMessage) && (
```

Comment:
```suggestion
      {!isStandalone && (helpText || countCharacters || validationMessage) && (
        <>
```

you can do this ☝️  and eliminate the second "if"